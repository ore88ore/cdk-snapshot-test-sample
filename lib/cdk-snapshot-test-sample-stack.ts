import { CfnOutput, RemovalPolicy, Stack, StackProps } from "aws-cdk-lib";
import { Construct } from "constructs";
import { NodejsFunction } from "aws-cdk-lib/aws-lambda-nodejs";
import * as dynamodb from "aws-cdk-lib/aws-dynamodb";
import { BillingMode } from "aws-cdk-lib/aws-dynamodb";
import * as apigateway from "aws-cdk-lib/aws-apigateway";

export class CdkSnapshotTestSampleStack extends Stack {
  constructor(scope: Construct, id: string, props?: StackProps) {
    super(scope, id, props);

    // DynamoDB
    const sampleTable = new dynamodb.Table(this, "sampleTable", {
      partitionKey: { name: "id", type: dynamodb.AttributeType.STRING },
      billingMode: BillingMode.PAY_PER_REQUEST,
      removalPolicy: RemovalPolicy.DESTROY,
    });
    new CfnOutput(this, "sampleTableName", {
      value: sampleTable.tableName,
      exportName: "sampleTableName",
    });

    // Lambda
    const sampleLambda = new NodejsFunction(this, "sampleLambda", {
      entry: "lib/sample-lambda.handler.ts",
      environment: {
        TABLE_NAME: sampleTable.tableName,
      },
    });
    sampleTable.grantReadData(sampleLambda);

    // API Gateway
    const sampleApi = new apigateway.RestApi(this, "sample-api");
    sampleApi.root.addMethod(
      "GET",
      new apigateway.LambdaIntegration(sampleLambda)
    );
  }
}
