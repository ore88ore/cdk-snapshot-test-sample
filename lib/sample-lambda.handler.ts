import { APIGatewayProxyEvent, APIGatewayProxyResult } from "aws-lambda";
import { DocumentClient } from "aws-sdk/clients/dynamodb";

export const handler = async (
  event: APIGatewayProxyEvent
): Promise<APIGatewayProxyResult> => {
  const client = new DocumentClient({ region: "ap-northeast-1" });
  const res = await client
    .scan({
      TableName: process.env.TABLE_NAME as string,
    })
    .promise();
  const responseBody = {
    count: res.Count ?? 0,
    samples: res.Items ?? [],
  };

  return {
    statusCode: 200,
    body: JSON.stringify(responseBody),
  };
};
