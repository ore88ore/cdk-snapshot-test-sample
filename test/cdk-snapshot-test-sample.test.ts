import * as cdk from "aws-cdk-lib";
import { Template } from "aws-cdk-lib/assertions";
import { CdkSnapshotTestSampleStack } from "../lib/cdk-snapshot-test-sample-stack";

test("snapshot test", () => {
  const app = new cdk.App();
  const stack = new CdkSnapshotTestSampleStack(app, "MyTestStack");
  const template = Template.fromStack(stack).toJSON();

  expect(template).toMatchSnapshot();
});
