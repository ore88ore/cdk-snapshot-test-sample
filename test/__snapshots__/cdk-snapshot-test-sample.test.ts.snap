// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot test 1`] = `
Object {
  "Outputs": Object {
    "sampleTableName": Object {
      "Export": Object {
        "Name": "sampleTableName",
      },
      "Value": Object {
        "Ref": "sampleTable0D61001F",
      },
    },
    "sampleapiEndpoint5691AF90": Object {
      "Value": Object {
        "Fn::Join": Array [
          "",
          Array [
            "https://",
            Object {
              "Ref": "sampleapiAA5ED7E4",
            },
            ".execute-api.",
            Object {
              "Ref": "AWS::Region",
            },
            ".",
            Object {
              "Ref": "AWS::URLSuffix",
            },
            "/",
            Object {
              "Ref": "sampleapiDeploymentStageprod1B60AC82",
            },
            "/",
          ],
        ],
      },
    },
  },
  "Parameters": Object {
    "BootstrapVersion": Object {
      "Default": "/cdk-bootstrap/hnb659fds/version",
      "Description": "Version of the CDK Bootstrap resources in this environment, automatically retrieved from SSM Parameter Store. [cdk:skip]",
      "Type": "AWS::SSM::Parameter::Value<String>",
    },
  },
  "Resources": Object {
    "sampleLambdaBA9DE42C": Object {
      "DependsOn": Array [
        "sampleLambdaServiceRoleDefaultPolicy8C711846",
        "sampleLambdaServiceRole07311550",
      ],
      "Properties": Object {
        "Code": Object {
          "S3Bucket": Object {
            "Fn::Sub": "cdk-hnb659fds-assets-\${AWS::AccountId}-\${AWS::Region}",
          },
          "S3Key": "[HASH REMOVED]",
        },
        "Environment": Object {
          "Variables": Object {
            "AWS_NODEJS_CONNECTION_REUSE_ENABLED": "1",
            "TABLE_NAME": Object {
              "Ref": "sampleTable0D61001F",
            },
          },
        },
        "Handler": "index.handler",
        "Role": Object {
          "Fn::GetAtt": Array [
            "sampleLambdaServiceRole07311550",
            "Arn",
          ],
        },
        "Runtime": "nodejs14.x",
      },
      "Type": "AWS::Lambda::Function",
    },
    "sampleLambdaServiceRole07311550": Object {
      "Properties": Object {
        "AssumeRolePolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": Object {
                "Service": "lambda.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "ManagedPolicyArns": Array [
          Object {
            "Fn::Join": Array [
              "",
              Array [
                "arn:",
                Object {
                  "Ref": "AWS::Partition",
                },
                ":iam::aws:policy/service-role/AWSLambdaBasicExecutionRole",
              ],
            ],
          },
        ],
      },
      "Type": "AWS::IAM::Role",
    },
    "sampleLambdaServiceRoleDefaultPolicy8C711846": Object {
      "Properties": Object {
        "PolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": Array [
                "dynamodb:BatchGetItem",
                "dynamodb:GetRecords",
                "dynamodb:GetShardIterator",
                "dynamodb:Query",
                "dynamodb:GetItem",
                "dynamodb:Scan",
                "dynamodb:ConditionCheckItem",
              ],
              "Effect": "Allow",
              "Resource": Array [
                Object {
                  "Fn::GetAtt": Array [
                    "sampleTable0D61001F",
                    "Arn",
                  ],
                },
                Object {
                  "Ref": "AWS::NoValue",
                },
              ],
            },
          ],
          "Version": "2012-10-17",
        },
        "PolicyName": "sampleLambdaServiceRoleDefaultPolicy8C711846",
        "Roles": Array [
          Object {
            "Ref": "sampleLambdaServiceRole07311550",
          },
        ],
      },
      "Type": "AWS::IAM::Policy",
    },
    "sampleTable0D61001F": Object {
      "DeletionPolicy": "Delete",
      "Properties": Object {
        "AttributeDefinitions": Array [
          Object {
            "AttributeName": "id",
            "AttributeType": "S",
          },
        ],
        "BillingMode": "PAY_PER_REQUEST",
        "KeySchema": Array [
          Object {
            "AttributeName": "id",
            "KeyType": "HASH",
          },
        ],
      },
      "Type": "AWS::DynamoDB::Table",
      "UpdateReplacePolicy": "Delete",
    },
    "sampleapiAA5ED7E4": Object {
      "Properties": Object {
        "Name": "sample-api",
      },
      "Type": "AWS::ApiGateway::RestApi",
    },
    "sampleapiAccount074F5E10": Object {
      "DependsOn": Array [
        "sampleapiAA5ED7E4",
      ],
      "Properties": Object {
        "CloudWatchRoleArn": Object {
          "Fn::GetAtt": Array [
            "sampleapiCloudWatchRole14C66026",
            "Arn",
          ],
        },
      },
      "Type": "AWS::ApiGateway::Account",
    },
    "sampleapiCloudWatchRole14C66026": Object {
      "Properties": Object {
        "AssumeRolePolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": Object {
                "Service": "apigateway.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "ManagedPolicyArns": Array [
          Object {
            "Fn::Join": Array [
              "",
              Array [
                "arn:",
                Object {
                  "Ref": "AWS::Partition",
                },
                ":iam::aws:policy/service-role/AmazonAPIGatewayPushToCloudWatchLogs",
              ],
            ],
          },
        ],
      },
      "Type": "AWS::IAM::Role",
    },
    "sampleapiDeployment9D4ABE32a81615675f70911410286643a5514148": Object {
      "DependsOn": Array [
        "sampleapiGET5F5EB503",
      ],
      "Properties": Object {
        "Description": "Automatically created by the RestApi construct",
        "RestApiId": Object {
          "Ref": "sampleapiAA5ED7E4",
        },
      },
      "Type": "AWS::ApiGateway::Deployment",
    },
    "sampleapiDeploymentStageprod1B60AC82": Object {
      "Properties": Object {
        "DeploymentId": Object {
          "Ref": "sampleapiDeployment9D4ABE32a81615675f70911410286643a5514148",
        },
        "RestApiId": Object {
          "Ref": "sampleapiAA5ED7E4",
        },
        "StageName": "prod",
      },
      "Type": "AWS::ApiGateway::Stage",
    },
    "sampleapiGET5F5EB503": Object {
      "Properties": Object {
        "AuthorizationType": "NONE",
        "HttpMethod": "GET",
        "Integration": Object {
          "IntegrationHttpMethod": "POST",
          "Type": "AWS_PROXY",
          "Uri": Object {
            "Fn::Join": Array [
              "",
              Array [
                "arn:",
                Object {
                  "Ref": "AWS::Partition",
                },
                ":apigateway:",
                Object {
                  "Ref": "AWS::Region",
                },
                ":lambda:path/2015-03-31/functions/",
                Object {
                  "Fn::GetAtt": Array [
                    "sampleLambdaBA9DE42C",
                    "Arn",
                  ],
                },
                "/invocations",
              ],
            ],
          },
        },
        "ResourceId": Object {
          "Fn::GetAtt": Array [
            "sampleapiAA5ED7E4",
            "RootResourceId",
          ],
        },
        "RestApiId": Object {
          "Ref": "sampleapiAA5ED7E4",
        },
      },
      "Type": "AWS::ApiGateway::Method",
    },
    "sampleapiGETApiPermissionMyTestStacksampleapiCE427428GET223FF364": Object {
      "Properties": Object {
        "Action": "lambda:InvokeFunction",
        "FunctionName": Object {
          "Fn::GetAtt": Array [
            "sampleLambdaBA9DE42C",
            "Arn",
          ],
        },
        "Principal": "apigateway.amazonaws.com",
        "SourceArn": Object {
          "Fn::Join": Array [
            "",
            Array [
              "arn:",
              Object {
                "Ref": "AWS::Partition",
              },
              ":execute-api:",
              Object {
                "Ref": "AWS::Region",
              },
              ":",
              Object {
                "Ref": "AWS::AccountId",
              },
              ":",
              Object {
                "Ref": "sampleapiAA5ED7E4",
              },
              "/",
              Object {
                "Ref": "sampleapiDeploymentStageprod1B60AC82",
              },
              "/GET/",
            ],
          ],
        },
      },
      "Type": "AWS::Lambda::Permission",
    },
    "sampleapiGETApiPermissionTestMyTestStacksampleapiCE427428GETA9A13D87": Object {
      "Properties": Object {
        "Action": "lambda:InvokeFunction",
        "FunctionName": Object {
          "Fn::GetAtt": Array [
            "sampleLambdaBA9DE42C",
            "Arn",
          ],
        },
        "Principal": "apigateway.amazonaws.com",
        "SourceArn": Object {
          "Fn::Join": Array [
            "",
            Array [
              "arn:",
              Object {
                "Ref": "AWS::Partition",
              },
              ":execute-api:",
              Object {
                "Ref": "AWS::Region",
              },
              ":",
              Object {
                "Ref": "AWS::AccountId",
              },
              ":",
              Object {
                "Ref": "sampleapiAA5ED7E4",
              },
              "/test-invoke-stage/GET/",
            ],
          ],
        },
      },
      "Type": "AWS::Lambda::Permission",
    },
  },
  "Rules": Object {
    "CheckBootstrapVersion": Object {
      "Assertions": Array [
        Object {
          "Assert": Object {
            "Fn::Not": Array [
              Object {
                "Fn::Contains": Array [
                  Array [
                    "1",
                    "2",
                    "3",
                    "4",
                    "5",
                  ],
                  Object {
                    "Ref": "BootstrapVersion",
                  },
                ],
              },
            ],
          },
          "AssertDescription": "CDK bootstrap stack version 6 required. Please run 'cdk bootstrap' with a recent version of the CDK CLI.",
        },
      ],
    },
  },
}
`;
